apps/src/JSDebuggerUI.js

```diff
@@ -0,0 +1,502 @@
+  if (slider) {
+    var sliderXOffset = 10,
+        sliderYOffset = 22,
+        sliderWidth = 130;
+    this.speedSlider_ = new Slider(sliderXOffset, sliderYOffset, sliderWidth,
+        slider);
+
+    // Change default speed (eg Speed up levels that have lots of steps).
+    if (options.defaultStepSpeedPercent) {
+      this.setStepSpeedPercent(options.defaultStepSpeedPercent);
+    }
+  }
+
+  // Attach keydown handler for debug console input area
+  var debugInput = this.rootDiv_.querySelector('#debug-input');
+  if (debugInput) {
+    debugInput.addEventListener('keydown', this.onDebugInputKeyDown.bind(this));
+  }
+
+  // Attach handlers for the debug area resize control
+  var resizeBar = this.getElement_('#debugResizeBar');
+  if (resizeBar) {
+    dom.addMouseDownTouchEvent(resizeBar, this.onMouseDownDebugResizeBar.bind(this));
+
+    // Can't use dom.addMouseUpTouchEvent() because it will preventDefault on
+    // all touchend events on the page, breaking click events...
+    document.body.addEventListener('mouseup',
+        this.onMouseUpDebugResizeBar.bind(this));
+    var mouseUpTouchEventName = dom.getTouchEventName('mouseup');
+    if (mouseUpTouchEventName) {
+      document.body.addEventListener(mouseUpTouchEventName,
+          this.onMouseUpDebugResizeBar.bind(this));
+    }
+  }
+
+  // Attach handler for console clear button
+  var clearButton = this.rootDiv_.querySelector('#clear-console-header');
+  if (clearButton) {
+    dom.addClickTouchEvent(clearButton, this.clearDebugOutput.bind(this));
+  }
+
+  // Attach handlers for debugger controls
+  var pauseButton = this.getElement_('#pauseButton');
+  var continueButton = this.getElement_('#continueButton');
+  var stepInButton = this.getElement_('#stepInButton');
+  var stepOverButton = this.getElement_('#stepOverButton');
+  var stepOutButton = this.getElement_('#stepOutButton');
+  if (pauseButton && continueButton && stepInButton && stepOverButton && stepOutButton) {
+    dom.addClickTouchEvent(pauseButton, this.onPauseContinueButton.bind(this));
+    dom.addClickTouchEvent(continueButton, this.onPauseContinueButton.bind(this));
+    dom.addClickTouchEvent(stepInButton, this.onStepInButton.bind(this));
+    dom.addClickTouchEvent(stepOverButton, this.onStepOverButton.bind(this));
+    dom.addClickTouchEvent(stepOutButton, this.onStepOutButton.bind(this));
+  }
+};
+
+/**
+ * Get the step delay in milliseconds from the speed slider in the debugger UI.
+ * If no speed slider is present, returns undefined.
+ * @return {number|undefined}
+ */
+JSDebuggerUI.prototype.getStepDelay = function () {
+  if (this.speedSlider_) {
+    return JSDebuggerUI.stepDelayFromSliderPercent(this.speedSlider_.getValue());
+  }
+  return undefined;
+};
+
+/**
+ * Set the speed slider position.
+ * @param {!number} percent - range 0.0-1.0
+ */
+JSDebuggerUI.prototype.setStepSpeedPercent = function (percent) {
+  if (this.speedSlider_) {
+    this.speedSlider_.setValue(percent);
+  }
+};
```

Comment:
Nice extraction, however "percent" doesn't seem right for the method or parameter names now that the input is between 0 and 1